mongodbatlas/config.go

```diff
@@ -55,21 +62,41 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 		return nil, diag.FromErr(err)
 	}
 
+	versionedClient, err := newVersionedClient(client)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
 	clients := &MongoDBClient{
-		Atlas:  atlasClient,
-		Config: c,
+		Atlas:    atlasClient,
+		AtlasSDK: versionedClient,
+		Config:   c,
 	}
 
 	return clients, nil
 }
 
+func newVersionedClient(client *http.Client) (*atlasSDK.APIClient, error) {
+	opts := []atlasSDK.ClientModifier{
+		atlasSDK.UseHTTPClient(client),
+		atlasSDK.UseUserAgent(userAgent)}
+
+	// Initialize the MongoDB Versioned Atlas Client.
+	versionedClient, err := atlasSDK.NewClient(opts...)
```

Comment:
Nit: Versioned is our internal term/feature of v2 api. Not user facing
```suggestion
	sdkv2, err := atlasSDK.NewClient(opts...)
```